src/rules/extensions.js

```diff
@@ -37,15 +39,15 @@ module.exports = function (context) {
     const extension = path.extname(resolvedPath || importPath).substring(1)
 
     if (!extension || !endsWith(importPath, extension)) {
-      if (isUseOfExtensionEnforced(extension)) {
+      if (isUseOfExtensionRequired(extension) && !isUseOfExtensionForbidden(extension)) {
```

Comment:
I'm confused by that - dependencies are always better. In this case it's ensuring that `hasOwnProperty` can't be broken.

Removing the dependency will effectively mean inlining the code - what's the benefit except reduced deducing in people's `node_modules`?